### DISCUSSION TITLE
Wazuh API Error after update linux update

### DISCUSSION THREAD

**Discussion by Aghiad90** [wazuh/wazuh#13605] [Q&A] [answered]
Good day we have issue with wazuh after the server install some updates please check the below error 
any idea how to fix or where to start from ? 
![image](https://user-images.githubusercontent.com/23388761/170740181-89228a40-4064-4cc8-bb4f-6b4fe155c31b.png)

**Comment by sebastiandbustos** [comment]
Hello @Aghiad90,
Thank you for using Wazuh!!!
I'm assuming you had Wazuh 4.2.x installed, if not, please provide your Wazuh version.
It seems like you didn't disabled the Wazuh repo after installing it, you should disable the repo so it won't attempt to update when you run a S.O. update/upgrade, if this is the case your Wazuh app's version probably is behind your current Wazuh manager's version.
To solve this you may remove the Wazuh app from your Kibana and install it again in the current version, first you will have to check your current Wazuh manager's version with:
```
/var/ossec/bin/wazuh-control -info
```
Now we will proceed to remove the old version of the Wazuh App:
```
cd /usr/share/kibana/
 sudo -u kibana bin/kibana-plugin remove wazuh
```
Install the new Wazuh Kibana plugin. Replace the version accordingly with the version retrieved earlier if necessary:
```
cd /usr/share/kibana/
sudo -u kibana /usr/share/kibana/bin/kibana-plugin install https://packages.wazuh.com/4.x/ui/kibana/wazuh_kibana-4.2.7_7.10.2-1.zip
```
Check that the config file for the Wazuh App (`/usr/share/kibana/data/wazuh/config/wazuh.yml`) and the credentials in it are similar to this.
```
hosts:
  - default:
      url: https://localhost
      port: 55000
      username: wazuh-wui
      password: wazuh-wui
      run_as: false
```
Once complete, restart Kibana:
```
systemctl restart kibana
```

Please let me know.
Regards.

**Reply by Aghiad90** [reply]
Hello @sebastiandbustos  , thank you for your reply. when i did check the version it shows as 4.3.1 should i still continue the process of removing and reinstalling ? 
![image](https://user-images.githubusercontent.com/23388761/171494186-3494085c-9c70-42cd-8897-b2e8b4815dbc.png)

**Comment by sebastiandbustos** [comment]
Hello @Aghiad90,
Do you recall wich version did you have installed previous to this?  (I'm guessing 4.3.0 perhaps?)
If you did have 4.3.0 then you will have to reinstall the Wazuh app, Then probably your indexer or dashboard are behind.
Can you please provide the versions for Wazuh-Indexer and Wazuh-Dashboard? (`yum list installed | grep i wazuh`)   Do you have an all in one installation?
Regards.

**Reply by Aghiad90** [reply]
Hello @sebastiandbustos  , it does not seem that i can find wazuh-indexer ? i have ubuntu server so i ran (apt list --installed | grep wazuh )
is there anyway to reinstall the right version without losing any data ?
![image](https://user-images.githubusercontent.com/23388761/171673444-2b09164d-934f-437a-8088-f7a259757099.png)

Regards ,

**Comment by sebastiandbustos** [comment]
Hello @Aghiad90,
Seems like your system updated Wazuh from 4.2.x to 4.3.1 on the last update, if you check your Wazuh repo probably you will see the enabled field set to 1 (on /etc/apt/sources.list.d/wazuh.list), in this case your Elasticsearch may work with the current version, but the Wazuh App plugin should be re-installed.
First would be a good idea to check your current Elasticsearch/OpenDistro/Kibana versions, for this you can run `apt list --installed | grep elastic`. This will most likely show elasticsearch on 7.10.2 version and opendistroforelasticsearch package on 1.13.2 version.
If this is normal you can proceed with the first step to update the Wazuh app plugin:
- Remove the old Wazuh app plugin:
```
cd /usr/share/kibana/
sudo -u kibana bin/kibana-plugin remove wazuh
```
- Install the new version of the Wazuh app plugin:
```
cd /usr/share/kibana/
sudo -u kibana /usr/share/kibana/bin/kibana-plugin install https://packages.wazuh.com/4.x/ui/kibana/wazuh_kibana-4.3.1_7.10.2-1.zip
```
- Restart your Kibana service:
```
systemctl restart kibana
```
After this please test your Kibana console again.
Please let me know.
Regards.

**Reply by Aghiad90** [reply]
Hello @sebastiandbustos thank you for your instruction after reinstalling wazuh 
i got this error on health check page : 
![image](https://user-images.githubusercontent.com/23388761/172207248-389f9ca2-adfb-4406-a0f3-ba2290ba128b.png)
![image](https://user-images.githubusercontent.com/23388761/172207322-29567ef1-fd06-4c8a-9bf6-b0f118fe758d.png)

**Reply by sebastiandbustos** [reply]
Hello @Aghiad90 ,
The error seems to be caused by a syntax error on the app configuration file (ˋ/usr/share/kibana/data/wazuh/config/wazuh.ymlˋ), please check that the config file has the correct syntax, if you can provide the contents we can also help to review it.
Thank you.
Best Regards.

**Reply by Aghiad90** [reply]
Hi @sebastiandbustos  , this is what i have in the config file the rest is commented 
![image](https://user-images.githubusercontent.com/23388761/172440077-b0cbdfb1-dabd-4d24-b7ca-69e51a6f3aec.png)

**Comment by sebastiandbustos** [comment]
Hello @Aghiad90 ,
The values seem to be ok, can you please restart the Kibana servide and provide the output for the following command?:
```
journalctl -u kibana --no-pager
```
Let me know.
Regards.

**Reply by Aghiad90** [reply]
Hi @sebastiandbustos  please check the output of  journalctl -u kibana --no-pager
["error","elasticsearch","data"],"pid":57260,"message":"[cluster_block_exception]: index [.kibana_1] blocked by: [TOO_MANY_REQUESTS/12/disk usage exceeded flood-stage watermark, index has read-only-allow-delete block];"}
May 23 09:27:25 wazuh kibana[57260]: {"type":"log","@timestamp":"2022-05-23T15:27:25Z","tags":["error","elasticsearch","data"],"pid":57260,"message":"[cluster_block_exception]: index [.kibana_1] blocked by: [TOO_MANY_REQUESTS/12/disk usage exceeded flood-stage watermark, index has read-only-allow-delete block];"}
May 23 09:27:25 wazuh kibana[57260]: {"type":"log","@timestamp":"2022-05-23T15:27:25Z","tags":["error","elasticsearch","data"],"pid":57260,"message":"[cluster_block_exception]: index [.kibana_1] blocked by: [TOO_MANY_REQUESTS/12/disk usage exceeded flood-stage watermark, index has read-only-allow-delete block];"}
May 23 09:27:25 wazuh kibana[57260]: {"type":"log","@timestamp":"2022-05-23T15:27:25Z","tags":["error","elasticsearch","data"],"pid":57260,"message":"[cluster_block_exception]: index [.kibana_1] blocked by: 

`

**Reply by Aghiad90** [reply]
![image](https://user-images.githubusercontent.com/23388761/173125950-f32abbaf-32e6-4c30-b275-850ae999e566.png)

**Comment by Aghiad90** [comment]
Hi @sebastiandbustos  any update on my issue ?

**Comment by Aghiad90** [comment, marked as answer]
Hi , @sebastiandbustos   i end it up by removing wazuh-registry.json under /user/share/kibana/data/wazuh/config 
and now it's working 100%
